### ISSUE TITLE
`TabNav` component incorrect focus order

### ISSUE THREAD

**Issue by GlebIrovich** [primer/react#2855]
### Description

[TabNav](https://primer.style/react/TabNav#support) component has incorrect focus order. 

Current behaviour:
When "tabbing" latest selected tab is focused

Expecte:
Active tab if focused. 
[Accessible example](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/examples/tabs-manual/)

https://user-images.githubusercontent.com/33176414/216620517-dc33c98f-f618-4226-b34a-347967e315e0.mov

### Steps to reproduce

1. Go to [docs](https://primer.style/react/TabNav#support)
2. Tab to TabNav, use arrow keys to select a tab which is not currently active
3. Tab back and forth 

### Version

v35.18.0

### Browser

Chrome

**Comment by rezrah** [comment]
Likely due to [tabIndex being set to -1](https://github.com/primer/react/blob/3bf1b0e368df10180291de205573df6c85434343/src/TabNav.tsx#L82). Unsure on reason for this, but it could be intentional. If we're happy to remove this however, it should resolve the issue.

**Comment by kendallgassner** [comment]
@inkblotty should we move this the audit-issues repo?

**Comment by inkblotty** [comment]
@kendallgassner - Yes that’s a good idea. We can add the right labels from
there.

**Comment by lesliecdubs** [comment]
Discussed with the PRC crew and decided we should use custom strategy of focus zone here. https://github.com/primer/behaviors/blob/main/docs/focus-zone.md#:~:text=from%20the%20first-,focusInStrategy,-%22first%22%20%E2%94%82%20%22closest%22%20%E2%94%82%20%22previous

**Comment by kendallgassner** [comment]
I cannot move this issue 🤔 to our audits repo. I wonder if its an `org` issue?

**Comment by kendallgassner** [comment]
I can make a corresponding ticket 😄

**Comment by joshblack** [comment]
Closed as part of: https://github.com/primer/react/pull/2845

This should be included in the latest release! Let me know if you're still seeing this issue 😄
